src/infrastructure/storage.py

```
@@ -0,0 +1,179 @@
+        self._csv_writer = None
+        self._lock = asyncio.Lock()
+        self._measurement_count = 0
+        self._header_written = False
+
+    async def __aenter__(self):
+        self.output_file.parent.mkdir(parents=True, exist_ok=True)
+        self.checkpoint_dir.mkdir(parents=True, exist_ok=True)
+        
+        mode = 'a' if self.output_file.exists() else 'w'
+        self._file_handle = await aiofiles.open(self.output_file, mode=mode, newline='')
+        
```

Pylint (3.3.7)

[convention] Trailing whitespace

(C0303)